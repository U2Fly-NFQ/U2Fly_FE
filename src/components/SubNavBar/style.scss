@use '../../assets/scss/abstracts' as a;

.sub-nav {
  display: flex;
  align-items: center;
  padding: 10px;
  border-bottom: a.$border-dark;
  height: 45px;

  &-list {
    display: flex;

    &__item {
      position: relative;
      list-style-type: none;
      display: inline-block;
      color: a.$white-color;
      margin-right: 25px;
      font-size: 14px;

      &::before {
        content: '';
        position: absolute;
        top: 3px;
        right: -15px;
        width: 1px;
        height: 15px;
        background-color: #dbdbdb78;
        z-index: 9999;
      }

      &:last-of-type::before {
        background-color: transparent;
      }

      a {
        padding-right: 7px;
        line-height: 20px;
      }
    }

    &-actions {
      font-weight: 500;
    }
  }
}

@media screen and (max-width: 1024px) {
  .sub-nav {
    background-color: a.$main-color;
  }
}

/* Extra small devices (phones, 600px and down) */
@media only screen and (max-width: 768px) {
  .sub-nav {
    .ant-row {
      justify-content: flex-end;
    }
  }
}
