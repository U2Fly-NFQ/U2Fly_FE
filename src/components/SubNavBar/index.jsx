import { Row, Col, Select } from 'antd'
import { NavLink } from 'react-router-dom'
import { useTranslation } from 'react-i18next'
import { DownOutlined } from '@ant-design/icons'
import { useNavigate } from 'react-router-dom'

import './style.scss'
import LangSelect from '../LangSelect'

const { Option } = Select

const SubNavBar = () => {
  const { t } = useTranslation()
  const user = JSON.parse(localStorage.getItem('user') || '[]')
  const navigate = useNavigate()

  return (
    <div className="sub-nav">
      <nav className="grid wide">
        <Row justify="space-between" align="middle" gutter={[2, 2]}>
          <Col md={12} sm={0}>
            <ul className="sub-nav-list">
              <li className="sub-nav-list__item">
                <a href="#">
                  <i className="fa-brands fa-facebook-square"></i>
                </a>
                <a href="">
                  <i className="fa-brands fa-twitter-square"></i>
                </a>
                <a href="">
                  <i className="fa-brands fa-instagram"></i>
                </a>
                <a href="">
                  <i className="fa-brands fa-linkedin"></i>
                </a>
              </li>
              <li className="sub-nav-list__item">+84 999 999 999</li>
              <li className="sub-nav-list__item">contact@domain.com</li>
            </ul>
          </Col>
          <Col flex="none">
            <ul className="sub-nav-list sub-nav-list-actions">
              {(Object.keys(user).length && (
                <li className="sub-nav-list__item">
                  <Select
                    className="lang-select"
                    size="small"
                    suffixIcon={
                      <DownOutlined
                        style={{
                          color: '#fff',
                        }}
                      />
                    }
                    defaultValue={user.username}
                    onChange={(value) => {
                      if (value === 'booked') navigate('/user/booked')
                      if (value === 'logout') {
                        localStorage.removeItem('user')
                        localStorage.removeItem('token')
                        navigate('/')
                      }
                    }}
                    styles={{
                      menuPortal: (base) => ({ ...base, zIndex: 9999 }),
                    }}
                  >
                    <Option key={'booked'}>My Booking</Option>
                    <Option key={'logout'}>Log out</Option>
                  </Select>
                </li>
              )) || (
                <>
                  <li className="sub-nav-list__item">
                    <NavLink to={'/login'}>{t('cta.login')}</NavLink>
                  </li>
                  <li className="sub-nav-list__item">
                    <NavLink to={'/register'}>{t('cta.register')}</NavLink>
                  </li>
                </>
              )}
              <li className="sub-nav-list__item">
                <LangSelect />
              </li>
            </ul>
          </Col>
        </Row>
      </nav>
    </div>
  )
}

export default SubNavBar
