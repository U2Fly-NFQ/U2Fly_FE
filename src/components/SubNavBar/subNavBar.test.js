import { I18nextProvider } from 'react-i18next'
import i18n from '../../translations'
import React, { Suspense } from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter as Router } from 'react-router-dom'
import { fireEvent, render, screen } from '@testing-library/react'
import SubNavBar from './'

// Storage Mock
const fakeLocalStorage = (function () {
  let store = {}

  return {
    getItem: function (key) {
      return store[key] || null
    },
    setItem: function (key, value) {
      store[key] = value.toString()
    },
    removeItem: function (key) {
      delete store[key]
    },
    clear: function () {
      store = {}
    },
  }
})()
describe('Sub nav bar test', () => {
  beforeAll(() => {
    Object.defineProperty(window, 'localStorage', {
      value: fakeLocalStorage,
    })
  })
  it('renders sub correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Suspense>
            <I18nextProvider i18n={i18n}>
              <SubNavBar />
            </I18nextProvider>
          </Suspense>
        </Router>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly', async () => {
    localStorage.setItem(
      'user',
      JSON.stringify({
        id: 1,
      })
    )
    render(
      <Router>
        <Suspense>
          <I18nextProvider i18n={i18n}>
            <SubNavBar />
          </I18nextProvider>
        </Suspense>
      </Router>
    )
    const infoProfile = screen.getByTestId('booking-profile')
    const listBox = screen.getByText('career@nfq.asia')
    fireEvent.click(infoProfile)
    fireEvent.click(listBox)
    // userEvent.click(screen.getByTestId('bookingProflie'))
  })
})
