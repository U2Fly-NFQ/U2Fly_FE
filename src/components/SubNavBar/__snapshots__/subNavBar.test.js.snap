// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sub nav bar test renders correctly 1`] = `
<div
  className="sub-nav"
>
  <nav
    className="grid wide"
  >
    <div
      className="ant-row ant-row-space-between ant-row-middle"
      style={
        Object {
          "marginBottom": -1,
          "marginLeft": -1,
          "marginRight": -1,
          "marginTop": -1,
        }
      }
    >
      <div
        className="ant-col ant-col-xs-0 ant-col-md-12"
        style={
          Object {
            "paddingBottom": 1,
            "paddingLeft": 1,
            "paddingRight": 1,
            "paddingTop": 1,
          }
        }
      >
        <ul
          className="sub-nav-list"
        >
          <li
            className="sub-nav-list__item"
          >
            <a
              href="#"
            >
              028 6681 2733
            </a>
          </li>
          <li
            className="sub-nav-list__item"
          >
            <a
              href="#"
            >
              career@nfq.asia
            </a>
          </li>
        </ul>
      </div>
      <div
        className="ant-col"
        style={
          Object {
            "flex": "none",
            "paddingBottom": 1,
            "paddingLeft": 1,
            "paddingRight": 1,
            "paddingTop": 1,
          }
        }
      >
        <ul
          className="sub-nav-list sub-nav-list-actions"
        >
          <li
            className="sub-nav-list__item"
          >
            <a
              className=""
              href="/login"
              onClick={[Function]}
            >
              Login
            </a>
          </li>
          <li
            className="sub-nav-list__item"
          >
            <a
              className=""
              href="/register"
              onClick={[Function]}
            >
              Register
            </a>
          </li>
          <li
            className="sub-nav-list__item"
          >
            <div
              className="ant-select ant-select-sm lang-select ant-select-single ant-select-show-arrow"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              styles={
                Object {
                  "menuPortal": [Function],
                }
              }
            >
              <div
                className="ant-select-selector"
                onClick={[Function]}
                onMouseDown={[Function]}
              >
                <span
                  className="ant-select-selection-search"
                >
                  <input
                    aria-activedescendant="undefined_list_0"
                    aria-autocomplete="list"
                    aria-controls="undefined_list"
                    aria-haspopup="listbox"
                    aria-owns="undefined_list"
                    autoComplete="off"
                    className="ant-select-selection-search-input"
                    disabled={false}
                    onChange={[Function]}
                    onCompositionEnd={[Function]}
                    onCompositionStart={[Function]}
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    onPaste={[Function]}
                    readOnly={true}
                    role="combobox"
                    style={
                      Object {
                        "opacity": 0,
                      }
                    }
                    type="search"
                    unselectable="on"
                    value=""
                  />
                </span>
                <span
                  className="ant-select-selection-item"
                >
                  <img
                    alt=""
                    src="united-kingdom.png"
                    style={
                      Object {
                        "marginBottom": "2px",
                        "marginRight": "4px",
                      }
                    }
                  />
                  ENG
                </span>
              </div>
              <span
                aria-hidden={true}
                className="ant-select-arrow"
                onMouseDown={[Function]}
                style={
                  Object {
                    "WebkitUserSelect": "none",
                    "userSelect": "none",
                  }
                }
                unselectable="on"
              >
                <span
                  aria-label="down"
                  className="anticon anticon-down"
                  role="img"
                  style={
                    Object {
                      "color": "#fff",
                    }
                  }
                >
                  <svg
                    aria-hidden="true"
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
